#!/bin/sh -e
## DP: Pluto debug logging patch

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -ur -x '*.svn*' icewm-1.2.21pre1.orig/src/misc.cc icewm-1.2.21pre1.patched/src/misc.cc
--- icewm-1.2.21pre1.orig/src/misc.cc	2006-07-24 18:51:14.000000000 +0300
+++ icewm-1.2.21pre1.patched/src/misc.cc	2006-08-31 20:15:30.000000000 +0300
@@ -15,6 +15,20 @@
 #include <libgen.h>
 #endif
 
+#include <sys/time.h>
+#include <time.h>
+void log_time_stamp()
+{
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    struct tm t;
+    localtime_r(&tv.tv_sec, &t);
+    char s[128];
+    memset(&s, 0, sizeof(s));
+    strftime(s, sizeof(s), "%H:%M:%S", &t);
+    fprintf(stderr, "%s.%06d ", s, tv.tv_usec);
+}
+
 extern char const *ApplicationName;
 
 #ifdef DEBUG
@@ -291,6 +305,7 @@
 }
 
 void precondition(char const *msg, ...) {
+    log_time_stamp();
     fprintf(stderr, "%s: ", ApplicationName);
 
     va_list ap;
@@ -303,6 +318,7 @@
 }
 
 void warn(char const *msg, ...) {
+    log_time_stamp();
     fprintf(stderr, "%s: ", ApplicationName);
     fputs(_("Warning: "), stderr);
 
@@ -315,6 +331,7 @@
 }
 
 void msg(char const *msg, ...) {
+    log_time_stamp();
     fprintf(stderr, "%s: ", ApplicationName);
 
     va_list ap;
